Build failure against OpenSSL 3.0

### Steps to reproduce

Run standard build against packaged OpenSSL 3.0.

### Expected behaviour

Successful build.

### Actual behaviour

Build failure:

```
FAILED: Telegram/CMakeFiles/Telegram.dir/SourceFiles/core/utils.cpp.o 
/usr/bin/g++ -DDESKTOP_APP_DISABLE_CRASH_REPORTS -DDESKTOP_APP_DISABLE_WAYLAND_INTEGRATION -DDESKTOP_APP_USE_PACKAGED -DDESKTOP_APP_USE_PACKAGED_FONTS -DG_LOG_DOMAIN=\"Telegram\" -DLOTTIE_USE_CACHE -DLOT_BUILD -DQT_CORE5COMPAT_LIB -DQT_CORE_LIB -DQT_DBUS_LIB -DQT_GUI_LIB -DQT_IMPLICIT_QCHAR_CONSTRUCTION -DQT_NETWORK_LIB -DQT_NO_CAST_FROM_BYTEARRAY -DQT_NO_DEBUG -DQT_NO_KEYWORDS -DQT_OPENGLWIDGETS_LIB -DQT_OPENGL_LIB -DQT_SVG_LIB -DQT_WIDGETS_LIB -DTDESKTOP_API_HASH=d524b414d21f4d37f08684c1df41ac9c -DTDESKTOP_API_ID=611335 -DTDESKTOP_DISABLE_AUTOUPDATE -DTDESKTOP_LAUNCHER_BASENAME=telegramdesktop -DTGCALLS_USE_STD_OPTIONAL -D_FORTIFY_SOURCE=2 -I/builddir/build/BUILD/tdesktop-3.2.4-full/redhat-linux-build/Telegram/Telegram_autogen/include -I/builddir/build/BUILD/tdesktop-3.2.4-full/Telegram/SourceFiles -I/builddir/build/BUILD/tdesktop-3.2.4-full/Telegram/lib_spellcheck -I/builddir/build/BUILD/tdesktop-3.2.4-full/Telegram/ThirdParty/tgcalls -I/builddir/build/BUILD/tdesktop-3.2.4-full/Telegram/ThirdParty/libtgvoip -I/builddir/build/BUILD/tdesktop-3.2.4-full/Telegram/lib_base -I/builddir/build/BUILD/tdesktop-3.2.4-full/Telegram/lib_rpl -I/builddir/build/BUILD/tdesktop-3.2.4-full/Telegram/lib_crl -I/builddir/build/BUILD/tdesktop-3.2.4-full/redhat-linux-build/Telegram/gen -I/builddir/build/BUILD/tdesktop-3.2.4-full/Telegram/lib_tl -I/builddir/build/BUILD/tdesktop-3.2.4-full/redhat-linux-build/Telegram/lib_ui/gen -I/builddir/build/BUILD/tdesktop-3.2.4-full/Telegram/lib_ui -I/builddir/build/BUILD/tdesktop-3.2.4-full/Telegram/lib_ui/emoji_suggestions -I/builddir/build/BUILD/tdesktop-3.2.4-full/Telegram/lib_lottie -I/builddir/build/BUILD/tdesktop-3.2.4-full/Telegram/lib_webrtc -I/builddir/build/BUILD/tdesktop-3.2.4-full/Telegram/lib_storage -I/builddir/build/BUILD/tdesktop-3.2.4-full/Telegram/lib_qr -I/builddir/build/BUILD/tdesktop-3.2.4-full/Telegram/lib_webview -I/builddir/build/BUILD/tdesktop-3.2.4-full/Telegram/SourceFiles/payments -I/builddir/build/BUILD/tdesktop-3.2.4-full/Telegram/ThirdParty/rlottie/inc -I/builddir/build/BUILD/tdesktop-3.2.4-full/Telegram/ThirdParty/statusnotifieritem -I/builddir/build/BUILD/tdesktop-3.2.4-full/Telegram/ThirdParty/libdbusmenu-qt/src -I/builddir/build/BUILD/tdesktop-3.2.4-full/redhat-linux-build/cmake/external/dbusmenu_qt -isystem /usr/include/minizip -isystem /usr/include/qt6/QtCore/6.2.1 -isystem /usr/include/qt6/QtCore/6.2.1/QtCore -isystem /usr/include/qt6/QtGui/6.2.1 -isystem /usr/include/qt6/QtGui/6.2.1/QtGui -isystem /usr/include/qt6/QtCore -isystem /usr/include/qt6 -isystem /usr/lib64/qt6/mkspecs/linux-g++ -isystem /usr/include/qt6/QtCore5Compat -isystem /usr/include/qt6/QtGui -isystem /usr/include/qt6/QtOpenGL -isystem /usr/include/qt6/QtWidgets -isystem /usr/include/qt6/QtOpenGLWidgets -isystem /usr/include/qt6/QtNetwork -isystem /usr/include/qt6/QtSvg -isystem /usr/include/qt6/QtDBus -isystem /usr/share -isystem /usr/include/giomm-2.4 -isystem /usr/lib64/giomm-2.4/include -isystem /usr/include/glib-2.0 -isystem /usr/lib64/glib-2.0/include -isystem /usr/include/sysprof-4 -isystem /usr/include/libmount -isystem /usr/include/blkid -isystem /usr/include/glibmm-2.4 -isystem /usr/lib64/glibmm-2.4/include -isystem /usr/include/sigc++-2.0 -isystem /usr/lib64/sigc++-2.0/include -isystem /usr/include/gio-unix-2.0 -isystem /usr/include/ffmpeg -isystem /usr/include/AL -isystem /usr/include/qrcodegencpp -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -DNDEBUG -fstack-protector-all -fstack-clash-protection -fPIC -fno-strict-aliasing -pipe -Wall -W -Wno-unused-parameter -Wno-switch -Wno-missing-field-initializers -Wno-sign-compare -Wno-maybe-uninitialized -Wno-deprecated-declarations -pthread -Winvalid-pch -include /builddir/build/BUILD/tdesktop-3.2.4-full/redhat-linux-build/Telegram/CMakeFiles/Telegram.dir/cmake_pch.hxx -MD -MT Telegram/CMakeFiles/Telegram.dir/SourceFiles/core/utils.cpp.o -MF Telegram/CMakeFiles/Telegram.dir/SourceFiles/core/utils.cpp.o.d -o Telegram/CMakeFiles/Telegram.dir/SourceFiles/core/utils.cpp.o -c /builddir/build/BUILD/tdesktop-3.2.4-full/Telegram/SourceFiles/core/utils.cpp
/builddir/build/BUILD/tdesktop-3.2.4-full/Telegram/SourceFiles/core/utils.cpp: In function 'void ThirdParty::finish()':
/builddir/build/BUILD/tdesktop-3.2.4-full/Telegram/SourceFiles/core/utils.cpp:128:17: error: 'FIPS_mode_set' was not declared in this scope; did you mean 'FIPS_mode'?
  128 |                 FIPS_mode_set(0);
      |                 ^~~~~~~~~~~~~
      |                 FIPS_mode
```

> The function calls 'FIPS_mode()' and 'FIPS_mode_set()' have been removed from OpenSSL 3.0. You should rewrite your application to not use them. See the sections below on how to write applications to use the FIPS Module in OpenSSL 3.0.

https://wiki.openssl.org/index.php/OpenSSL_3.0

### Operating system

Fedora Rawhide

### Version of Telegram Desktop

3.2.5 and latest master

### Installation source

Other (unofficial) source

### Logs

_No response_

--- a/Telegram/SourceFiles/core/utils.cpp
+++ b/Telegram/SourceFiles/core/utils.cpp
@@ -86,7 +86,11 @@ namespace ThirdParty {
 	}
 
 	void finish() {
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+		EVP_default_properties_enable_fips(nullptr, 0);
+#else
 		FIPS_mode_set(0);
+#endif
 		CONF_modules_unload(1);
 
 		Platform::ThirdParty::finish();

